modules/fedimint-ln-server/src/lib.rs

```diff
@@ -885,11 +919,60 @@ impl ServerModule for Lightning {
 }
 
 impl Lightning {
-    pub fn new(cfg: LightningConfig, task_group: &mut TaskGroup) -> anyhow::Result<Self> {
+    pub async fn new(
+        cfg: LightningConfig,
+        db: Database,
+        task_group: &mut TaskGroup,
+    ) -> anyhow::Result<Self> {
         let btc_rpc = create_bitcoind(&cfg.local.bitcoin_rpc, task_group.make_handle())?;
+        task_group
+            .spawn("compact audit keys", |handle| async move {
+                Self::compact_audit_keys_task(db, &handle).await;
+            })
+            .await;
         Ok(Lightning { cfg, btc_rpc })
     }
 
+    async fn compact_audit_keys_task(db: Database, tg_handle: &TaskHandle) {
+        let interval = Duration::from_secs(60 * 5);
+        while !tg_handle.is_shutting_down() {
+            let mut dbtx = db.begin_transaction().await;
```

Comment:
Can this possibly interfere with rest of the module and cause `commit_tx` failures elsewhere?

Reply:
Yes, this is a problem, we don't handle commit failures because we decided that consensus DB entries should not be written to concurrently. Could we instead just check after each `process_input` if a contract has become 0-value?

I'd even go a step further and require eventually that all funds be withdrawn at once.

Reply:
Oof, great catch! I've moved the logic to remove 0-value audit keys as the last step in `process_input`.

>I'd even go a step further and require eventually that all funds be withdrawn at once.

Eric, does this comment mean if there's a commit failure for consensus DB entries it would trigger a force withdrawal of all funds from the federation?